### PR TITLE
Improvement:  EKR multiturn conversation, and chat models added 

### PR DESCRIPTION
**Pull request by snova-jorgep**
refactoring ekr to use chat models
conversational qa chain with history using reformulated queries added

### CODE REVIEW

**Review by snova-kwasia — APPROVED** [review]
LGTM @snova-jorgep thanks!

File: enterprise_knowledge_retriever/src/document_retrieval.py [hunk 1]
```diff
@@ -239,54 +332,30 @@ def init_retriever(self, vectorstore: Any) -> None:
                 },
             )
 
-    def get_qa_retrieval_chain(self) -> RetrievalQAChain:
+    def get_qa_retrieval_chain(self, conversational: bool = False) -> RetrievalQAChain:
```
Comment (snova-petrojm) [inline enterprise_knowledge_retriever/src/document_retrieval.py:335]: Can we add 'conversational' as an input arg in config.yaml that users can control? Thanks

**Review by snova-petrojm — COMMENTED** [review]
Thanks @snova-jorgep. Let few comments. Also, can we add somewhere in document_retrieval.py a brief description of how multi-turn works with RAG when conversational is set to True? You said you will be 3 LLM calls, would be good to mention this somewhere.

File: enterprise_knowledge_retriever/src/document_retrieval.py (hunk 1 above)
Comment (snova-jorgep) [reply enterprise_knowledge_retriever/src/document_retrieval.py:341]: added under retrieval, given retrieval is done over reformulated or original query

**Pull request merged**